### PR TITLE
Move istio-kyma-patch logic to istio subchart

### PR DESCRIPTION
**Pull request by Demonsthere**
**Description**

Changes proposed in this pull request:

- Change isito-kyma-patch to be checking only
- Move isito-kyma-patch modifying/creating parts into new kyma subchart for istio 
- Update application-connector charts for istio sds

**Related issue(s)**

Implements #4252

### CODE REVIEW

File: resources/core/charts/gateway/templates/kyma-gateway-certs.yaml [hunk 1]
```diff
@@ -0,0 +1,10 @@
+---
+apiVersion: v1
+data:
+  "key": {{ .Values.global.ingress.tlsKey }}
+  "cert": {{ .Values.global.ingress.tlsCrt }}
+kind: Secret
+metadata:
+  name: {{ .Values.global.istio.gateway.name }}-certs
+  namespace: istio-system
```
Comment (strekm) [inline resources/core/charts/gateway/templates/kyma-gateway-certs.yaml]: remove namespace please

Comment (Demonsthere) [reply resources/core/charts/gateway/templates/kyma-gateway-certs.yaml]: Correction: the certs needs need to be in the same namespace that the ingress-gateway pod is

File: resources/istio/charts/kyma/templates/destination-rule.yaml [hunk 2]
```diff
@@ -1,7 +1,8 @@
+---
 apiVersion: networking.istio.io/v1alpha3
 kind: DestinationRule
 metadata:
-  name: istio-ingressgateway
+  name: istio-ingressgateway-dr
```
Comment (piotrmsc) [inline resources/istio/charts/kyma/templates/destination-rule.yaml]: nit: it is not needed to add suffixes with resource kind abbreviation when you fetch it you have to specify kind

File: resources/istio/charts/kyma/templates/label-job.yaml [hunk 3]
```diff
@@ -0,0 +1,50 @@
...
+  name: kyma-ns-label
+---
+apiVersion: rbac.authorization.k8s.io/v1
+kind: ClusterRole
+metadata:
+  name: kyma-ns-label
+rules:
+- apiGroups: [""]
+  resources: ["namespaces"]
+  verbs: ["get", "list", "watch", "patch"]
```
Comment (piotrmsc) [inline resources/istio/charts/kyma/templates/label-job.yaml]: Please limit the permissions to namespace it should work on it (use names from values file)

File: resources/istio/charts/kyma/templates/label-job.yaml [hunk 4]
```diff
@@ -0,0 +1,50 @@
...
+      restartPolicy: OnFailure
+      containers:
+      - name: kyma-ns-label
+        image: eu.gcr.io/kyma-project/test-infra/alpine-kubectl:v20190325-ff66a3a
+        command:
+          - /bin/bash
+          - -c
+          - |
+            for NS in istio-system kube-system; do
+              kubectl label namespace "${NS}" "istio-injection=disabled" --overwrite
```
Comment (piotrmsc) [inline resources/istio/charts/kyma/templates/label-job.yaml]: I suggest to make it configurable via values.yaml

File: resources/istio/charts/kyma/values.yaml [hunk 5]
```diff
@@ -0,0 +1,7 @@
+---
+enabled: false
```
Comment (piotrmsc) [inline resources/istio/charts/kyma/values.yaml]:
```suggestion
enabled: true
```

File: resources/istio/charts/kyma/Chart.yaml [hunk 6]
```diff
@@ -0,0 +1,6 @@
+apiVersion: v1
```
Comment (piotrmsc) [inline resources/istio/charts/kyma/Chart.yaml]: Maybe this should be a part of isito init an label namespace before anything is applied to istio system etc?
And please change name, it is confusing

File: resources/istio/charts/kyma/templates/label-job.yaml [hunk 7]
```diff
@@ -0,0 +1,50 @@
...
+  kind: ClusterRole
+  name: kyma-ns-label
+subjects:
+- kind: ServiceAccount
+  name: kyma-ns-label
+  namespace: istio-system
+---
+apiVersion: batch/v1
+kind: Job
+metadata:
```
Comment (piotrmsc) [inline resources/istio/charts/kyma/templates/label-job.yaml]: I would invoke this as a post install & upgrade step and add some logging there when it fails

File: resources/istio/charts/kyma/templates/label-job.yaml [hunk 8]
```diff
@@ -0,0 +1,50 @@
...
+  name: kyma-ns-label
+spec:
+  backoffLimit: 1
```
Comment (piotrmsc) [inline resources/istio/charts/kyma/templates/label-job.yaml]: bacoffLimit is set to 1 and you have restart policy on failure which is contrary. Default backoffLimit is 6 so I would either delete it or specify a different limit

File: resources/istio-kyma-patch/templates/configmap.yaml [hunk 9]
```diff
@@ -7,53 +7,5 @@ metadata:
...
-        }
-      }
-    ]
-
-  injection-in-namespaces: |
-    istio-system disabled
-    kube-system disabled
-
-  delete: |
-    deployment prometheus
```
Comment (piotrmsc) [inline resources/istio-kyma-patch/templates/configmap.yaml]: Some of those resources should be disabled in the charts

File: docs/application-connector/08-10-rotate-root-ca.md [hunk 10]
```diff
@@ -115,7 +115,7 @@ To successfully rotate a soon-to-expire CA certificate, replace it with a new ce
 
 5. Replace the old certificate in the Istio Secret. Edit the Secret and replace the `ca.crt` value with the new base64-encoded certificate. Run:
   ```
-  kubectl -n istio-system edit secret application-connector-ca-certs
+  kubectl -n istio-system edit secret application-connector-certs-cacert
   ```
 
 6. If you experience any issues with new certificates not being trusted, restart Istio Ingress Gateway Pods. Run:
```
Comment (Szymongib) [inline docs/application-connector/08-10-rotate-root-ca.md]: From what I understand after these changes, we no longer need to restart the pod after changing the content of the secret. Am I wrong on this?

File: resources/core/charts/gateway/templates/application-connector-certs.yaml [hunk 11]
```diff
@@ -1,7 +1,18 @@
...
+kind: Secret
+metadata:
+  name: application-connector-certs-cacert
+  namespace: istio-system
```
Comment (Szymongib) [inline resources/core/charts/gateway/templates/application-connector-certs.yaml]: Same here, please use `{{ .Values.global.istio.namespace }}`

File: resources/istio/charts/customization/values.yaml [hunk 12]
```diff
@@ -0,0 +1,3 @@
+---
+enabled: false
+namespaces2Label: []
```
Comment (piotrmsc) [inline resources/istio/charts/customization/values.yaml]: I would put the values here, it will be easier to upgrade charts (just copy paste sub chart)

**Pull request merged**